decidim-core/app/helpers/decidim/main_app_missing_routes_helper.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Decidim
+  # A helper to get paths from main_app routes if cannot be found in the
+  # current engine
+  module MainAppMissingRoutesHelper
```

Comment:
You are right, I think that this module is not required currently and may affect drastically to performance. In fact, in a previous commit, I had to revert the use of MainAppPolymorphicMappings in a middleware because the performance dropped a lot. I think that the current approach is safer.